### PR TITLE
Sanity check the size of the image buffer in C++

### PR DESCRIPTION
**Pull request by emilk**
### What
* Part of https://github.com/rerun-io/rerun/issues/7050

Sanity check the size of the image buffer in C++

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7163?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7163?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide

- [PR Build Summary](https://build.rerun.io/pr/7163)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: rerun_cpp/src/rerun/datatypes/image_format_ext.cpp [hunk 1]
```diff
@@ -41,6 +41,24 @@ namespace rerun {
...
 
+            /// How many bytes will this image occupy?
+            size_t num_bytes() const {
+                return width * height * this->bits_per_pixel() / 8;
+            }
+
+            /// How many bits per pixel?
+            ///
+            /// Note that this is not necessarily a factor of 8.
+            size_t bits_per_pixel() const {
```
Comment (Wumpf) [inline rerun_cpp/src/rerun/datatypes/image_format_ext.cpp:52]: Urgh, I dislike the notion of bits per pixel.
We had the same issue on wgpu where contributor wanted to have bytes per pixel but then block compression is a thing and the whole notion of pixel gets super messy - as is already with chroma downsampling! Most block compression formats still have non-fractional bits per decoded pixel, but at latest when ASTC's variable block size enters the picture everything breaks down.
I'd rather we not have this and instead have a method on pixel format that gives you the expected size for under a resolution + format. The model+datatype path is ofc trivial and can stay here.

**Review by Wumpf — APPROVED** [review]
error handling lgtm

won't die on the `bits_per_pixel` hill today, so not blocking over this (I'll reserve _told you so_ rights though if this goes in unchanged ;))

File: rerun_cpp/src/rerun/datatypes/image_format_ext.cpp (hunk 1 above)
Comment (emilk) [reply rerun_cpp/src/rerun/datatypes/image_format_ext.cpp:52]: fixed!

Comment (Wumpf) [reply rerun_cpp/src/rerun/datatypes/image_format_ext.cpp]: <3

**Review by Wumpf — APPROVED** [review]
thank you! :)

**Pull request merged**